src/vs/workbench/contrib/share/browser/share.contribution.ts

```diff
@@ -46,82 +47,95 @@ const targetMenus = [
 class ShareWorkbenchContribution {
 	private static SHARE_ENABLED_SETTING = 'workbench.experimental.share.enabled';
 
+	private _disposables = new DisposableStore();
+
 	constructor(
 		@IShareService private readonly shareService: IShareService,
 		@IConfigurationService private readonly configurationService: IConfigurationService
 	) {
 		if (this.configurationService.getValue<boolean>(ShareWorkbenchContribution.SHARE_ENABLED_SETTING)) {
 			this.registerActions();
 		}
+		this.configurationService.onDidChangeConfiguration(e => {
+			if (e.affectsConfiguration(ShareWorkbenchContribution.SHARE_ENABLED_SETTING)) {
+				if (this.configurationService.getValue<boolean>(ShareWorkbenchContribution.SHARE_ENABLED_SETTING)) {
+					this.registerActions();
```

Comment:
This could lead to registering the actions multiple times. You might want to clear the disposable first. Your setting value's type won't be validated, I could change it from `true` to "abc" and hit this line